### PR TITLE
feat: Option to download barcode

### PR DESCRIPTION
**Pull request by gitstart**
## Change Summary

fixes: #4814
option to download barcode

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)

## Test/ Verification

Provide summary of changes.
 Option to download barcode

## Additional information / screenshots (optional)
https://www.loom.com/share/29e2ba9d7f574320af2d4cb9b44e09b1

---
This code was written and reviewed by GitStart Community. Growing future engineers, one PR at a time.

### CODE REVIEW

**Review by wingkwong — CHANGES_REQUESTED** [review]
**1. Your changes will break the tests.**
 
Please update `tests/playwright/tests/columnBarcode.spec.ts`

**2. The barcode width looks quite different after the change.**

left: your PR, right: 0.104.3

image

image

image

image

image

**Comment by pranavxc** [comment]
> 2. The barcode width looks quite different after the change.

@wingkwong Looks like both images are identical but it's stretched to fit the cell. We have to keep the original image aspect ratio then it will be good.

File: packages/nc-gui/components/virtual-cell/barcode/JsBarcodeWrapper.vue [hunk 1]
```diff
@@ -42,6 +42,13 @@ onMounted(generate)
 </script>
 
 <template>
-  <svg v-show="!errorForCurrentInput" ref="barcodeSvgRef" class="w-full" data-testid="barcode" @click="onBarcodeClick"></svg>
+  <img
+    v-show="!errorForCurrentInput"
+    ref="barcodeImgRef"
+    alt="barcode"
+    class="w-full"
```
Comment (pranavxc) [inline packages/nc-gui/components/virtual-cell/barcode/JsBarcodeWrapper.vue:49]: The `w-full` is breaking the barcode image aspect ratio.

**Comment by wingkwong** [comment]
@pranavxc Yes. I'm talking about the width being stretched only. The barcode content itself is same.

**Comment by frankMagoba** [comment]
> **1. Your changes will break the tests.**
> 
> Please update `tests/playwright/tests/columnBarcode.spec.ts`
> 
> **2. The barcode width looks quite different after the change.**
> 
> left: your PR, right: 0.104.3
> 
> image
> 
> image
> 
> image
> 
> image
> 
> image

Hello @wingkwong , I was considering a way to allow for downloading the barcode svgs in light of the test cases being run against svg fixtures. My idea is to include an action button that triggers a download event. However, implementing this would require a recommendation for the UI design of the button used to download the barcodes.

**Comment by gitstart** [comment]
> **1. Your changes will break the tests.**
> 
> Please update `tests/playwright/tests/columnBarcode.spec.ts`
> 
> **2. The barcode width looks quite different after the change.**
> 
> left: your PR, right: 0.104.3
> 
> image
> 
> image
> 
> image
> 
> image
> 
> image

Hello @wingkwong , I was considering a way to allow for downloading the barcode svgs in light of the test cases being run against svg fixtures. My idea is to include an action button that triggers a download event. However, implementing this would require a recommendation for the UI design of the button used to download the barcodes.

**Comment by wingkwong** [comment]
You original change is fine. You just need to handle the CSS style issue.

**Comment by wingkwong** [comment]
Closing - inactivity.

**Pull request closed without merging**